main/kotlin/com/example/people_managment_system/service/SystemService.kt

```diff
@@ -0,0 +1,177 @@
+    Takes in the company name as a path variable.
+    Returns a String response if insufficient or erroneous data is entered. */
+
+    fun saveNewEmployee(companyName: String ,newEmployeeDto: EmployeeDto): String {
+       val newEmployee = Employee()
+       newEmployee.employeeID = newEmployeeDto.employeeID
+       newEmployee.employeeName = newEmployeeDto.employeeName
+       newEmployee.email = newEmployeeDto.email
+       newEmployee.salary = newEmployeeDto.salary
+       newEmployee.company = companyRepository.getCompanyByName(companyName)
+       newEmployee.manager = managerRepository.getManagerById(newEmployeeDto.managerID)
+       newEmployee.team = teamRepository.getTeamById(newEmployeeDto.teamID)
+       return try {
+           employeeRepository.save(newEmployee)
+           "The Employee has been successfully saved to PeopleMs"
+       } catch (e: DataIntegrityViolationException) {
+           throw DataIntegrityViolationException("Could not add the employee to your company. " +
+                   "Incorrect or insufficient data supplied.")
+       }
+    }
+
+/* Updates the Team of an employee.
+   Takes in the company name as a path variable to ensure the Team ID relates to the right company.
+   Throws an exception if the employee ID does not exist. */
+
+    fun updateTeam(companyName: String, relevantEmployeeId: Int, relevantTeamId: Int): EmployeeDto {
+       var relevantEmployee: Employee? = null
+       for (item in employeeRepository.findAll()) {
+           if (item.employeeID == relevantEmployeeId &&
+                   item.company?.companyName == companyName) {
+               relevantEmployee = item
+               relevantEmployee.team = teamRepository.getTeamById(relevantTeamId)
+               employeeRepository.save(relevantEmployee)
+           }
+       }
+       return relevantEmployee?.let { mapEmployeeEntityToEmployeeDto(it) }
+           ?: throw NoSuchElementException("Could not find an employee with that ID")
+    }
+
+/* Updates the Manager of an employee.
+   Takes in the company name as a path variable to ensure the Manager ID relates to the right company.
+   Throws an exception if the employee ID does not exist. */
+
+fun updateManager(companyName: String, relevantEmployeeId: Int, relevantManagerId: Int): EmployeeDto {
+   var relevantEmployee: Employee? = null
+   for (item in employeeRepository.findAll()) {
+       if (item.employeeID == relevantEmployeeId &&
+           item.company?.companyName == companyName) {
+           relevantEmployee = item
+           relevantEmployee.manager = managerRepository.getManagerById(relevantManagerId)
+           employeeRepository.save(relevantEmployee)
+       }
+   }
+   return relevantEmployee?.let { mapEmployeeEntityToEmployeeDto(it) }
+       ?: throw NoSuchElementException("Could not find an employee with that ID")
+}
+
+/* Deletes an employee's data.
+   Returns a String response if the employee ID does not exist. */

```

Comment:
Usually with a delete operation, you don't need to return anything, you can pass the ID to delete and just execute it, expect that once the ID is passed and executed, if it's there it's deleted if no exception is thrown and if it's not there then it's deleted anyway.